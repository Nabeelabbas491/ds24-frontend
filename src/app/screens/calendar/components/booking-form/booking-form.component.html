<div>
  <h4 class="font-medium">{{ 'BOOKING.BOOKING_FORM.ENTER_DETAILS' | translate }}</h4>
  <form [formGroup]="form" novalidate (submit)="confirmBooking()" #bookingForm="ngForm">
    <ds-input-wrapper class="mt-2 text-left">
      <input dsInput formControlName="name" type="text" [placeholder]="'BOOKING.BOOKING_FORM.NAME' | translate" />
      <ds-form-label>{{ 'BOOKING.BOOKING_FORM.NAME' | translate }}</ds-form-label>
    </ds-input-wrapper>
    <ds-form-sub-text *ngIf="leftCountName$ | async as leftCountName" class="text-left">
      {{ 'BOOKING.BOOKING_FORM.NAME_DESCRIPTION' | translate: { value: leftCountName } }}
    </ds-form-sub-text>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.name?.hasError('required') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.NAME_REQUIRED' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.name?.hasError('maxlength') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.NAME_MAX_LENGTH' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.name?.hasError('whitespace') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.ENTER_VALID_TEXT' | translate
      }}</ds-form-error>
    </div>
    <ds-input-wrapper class="mt-2 text-left">
      <input
        dsInput
        formControlName="email"
        type="email"
        [placeholder]="'BOOKING.BOOKING_FORM.EMAIL' | translate"
        email
      />
      <ds-form-label>{{ 'BOOKING.BOOKING_FORM.EMAIL' | translate }}</ds-form-label>
    </ds-input-wrapper>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.email?.hasError('required') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.EMAIL_REQUIRED' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.email?.hasError('maxlength') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.EMAIL_MAX_LENGTH' | translate
      }}</ds-form-error>
      <ds-form-error
        *ngIf="
          !formControls.email?.hasError('maxlength') && formControls.email?.hasError('email') && bookingForm.submitted
        "
        >{{ 'BOOKING.BOOKING_FORM.EMAIL_NOT_IN_VALID_FORMAT' | translate }}</ds-form-error
      >
    </div>
    <ng-container *ngIf="meetingType$ | async as meetingType">
      <ds-input-wrapper class="mt-2 text-left" *ngIf="meetingType.id === meetingTypeIds.Outbound">
        <input
          dsInput
          formControlName="phoneNo"
          type="text"
          [placeholder]="'BOOKING.BOOKING_FORM.PHONE_NUMBER' | translate"
        />
        <ds-form-label>{{ 'BOOKING.BOOKING_FORM.PHONE_NUMBER' | translate }}</ds-form-label>
      </ds-input-wrapper>
      <ds-form-error *ngIf="formControls.phoneNo.hasError('pattern')">
        <div class="pt-1">{{ 'BOOKING.BOOKING_FORM.INLCUDE_COUNTRY_CODE_TEXT' | translate }}</div>
      </ds-form-error>
    </ng-container>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.phoneNo?.hasError('required') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.PHONE_NUMBER_REQUIRED' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.phoneNo?.hasError('pattern') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.PHONE_NUMBER_NOT_IN_VALID_FORMAT' | translate
      }}</ds-form-error>
    </div>
    <ds-textarea class="mt-4 text-left">
      <textarea
        dsInput
        formControlName="note"
        class="h-[150px]"
        [placeholder]="'BOOKING.BOOKING_FORM.SHARE_TEXT' | translate"
      ></textarea>
      <ds-form-label>{{ 'BOOKING.BOOKING_FORM.NOTES' | translate }}</ds-form-label>
    </ds-textarea>
    <ds-form-sub-text *ngIf="leftCountComments$ | async as leftCountComments" class="text-left">
      {{ 'BOOKING.BOOKING_FORM.COMMENTS_DESCRIPTION' | translate: { value: leftCountComments } }}
    </ds-form-sub-text>
    <div class="w-full text-left">
      <ds-form-error *ngIf="formControls.note?.hasError('maxlength') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.COMMENTS_MAX_LENGTH' | translate
      }}</ds-form-error>
      <ds-form-error *ngIf="formControls.note?.hasError('whitespace') && bookingForm.submitted">{{
        'BOOKING.BOOKING_FORM.ENTER_VALID_TEXT' | translate
      }}</ds-form-error>
    </div>
    <div class="mt-5" *ngIf="isMobileView$ | async">
      <button
        dsButton="default"
        class="w-full h-10 py-2 px-4 shadow-shadow-z200 text-center"
        [disabled]="(saveBookingPending$ | async)!"
        (click)="prevStep()"
      >
        {{ 'BOOKING.BOOKING_OTHER.GO_BACK' | translate }}
      </button>
    </div>
    <div class="mt-5">
      <button
        class="w-full h-10 py-2 px-4 bg-primary-p300-main text-neutral-n0 shadow-shadow-z200 text-center"
        type="submit"
        [disabled]="(saveBookingPending$ | async)!"
      >
        {{ 'BOOKING.BOOKING_OTHER.CONFIRM_BOOKING' | translate }}
      </button>
    </div>
    <div class="w-full text-center mt-5">
      <ds-form-error *ngIf="error$ | async as errorMessage">{{ errorMessage }}</ds-form-error>
    </div>
  </form>
  <div class="flex flex-row justify-center">
    <ds-spinner *ngIf="saveBookingPending$ | async" [color]="spinnerColor" [size]="30"></ds-spinner>
  </div>
</div>
