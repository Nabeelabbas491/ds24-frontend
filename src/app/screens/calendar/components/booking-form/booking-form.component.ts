import { Component } from '@angular/core';
import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
import { ReactiveLifecycles } from '@ds24/utilities';
import { Store } from '@ngrx/store';
import { actions, selectors } from '../../../../store';
import { BehaviorSubject, filter, takeUntil } from 'rxjs';
import { MeetingTypeId } from './../../../../types/booking.type';
import { IconColor } from '@ds24/elements';
import { noWhitespaceValidator } from './../../../../shared/common/util';

@Component({
  selector: 'ds-booking-form',
  templateUrl: './booking-form.component.html',
})
export class BookingFormComponent extends ReactiveLifecycles {
  meetingTypeIds = MeetingTypeId;

  meetingType$ = this.store.select(selectors.booking.selectMeetingType);
  isMobileView$ = this.store.select(selectors.viewBreakPoint.selectIsMobileView);
  error$ = this.store.select(selectors.bookingAPI.selectBookingSaveError);
  saveBookingPending$ = this.store.select(selectors.bookingAPI.selectBookingSavePending);

  leftCountName$ = new BehaviorSubject<number>(80);
  leftCountComments$ = new BehaviorSubject<number>(300);

  spinnerColor = IconColor.Primary300;

  form: FormGroup = this.fb.group({
    name: new FormControl('', { validators: [Validators.required, Validators.maxLength(80), noWhitespaceValidator] }),
    email: new FormControl('', { validators: [Validators.required, Validators.maxLength(255)] }),
    phoneNo: new FormControl('', { validators: [Validators.pattern(/^\+\d{1,4}\s?\d{1,}$/)] }),
    note: new FormControl('', { validators: [Validators.maxLength(300), noWhitespaceValidator] }),
  });

  constructor(
    private fb: FormBuilder,
    private store: Store,
  ) {
    super();
  }

  ngOnInit() {
    super.ngOnInit();

    this.saveBookingPending$.pipe(takeUntil(this.ngOnDestroy$)).subscribe(isPending => {
      if (isPending) {
        this.form.disable();
      } else {
        this.form.enable();
      }
    });

    this.form
      .get('name')
      ?.valueChanges.pipe(takeUntil(this.ngOnDestroy$))
      .subscribe(value => {
        value ? this.leftCountName$.next(Math.max(0, 80 - value?.length)) : this.leftCountName$.next(80);
      });

    this.form
      .get('note')
      ?.valueChanges.pipe(takeUntil(this.ngOnDestroy$))
      .subscribe(value => {
        value ? this.leftCountComments$.next(Math.max(0, 300 - value?.length)) : this.leftCountComments$.next(300);
      });

    this.meetingType$
      .pipe(
        filter(meetingType => meetingType?.id === this.meetingTypeIds.Outbound),
        takeUntil(this.ngOnDestroy$),
      )
      .subscribe(() => {
        //eslint-disable-line
        this.formControls.phoneNo?.setValidators([Validators.required, Validators.pattern(/^\+\d{1,4}\s?\d{1,}$/)]);
      });
  }

  prevStep() {
    this.store.dispatch(actions.booking.selectStep({ step: 'booking_type' }));
  }
  confirmBooking() {
    this.form.markAllAsTouched();
    if (this.form.valid) {
      this.store.dispatch(actions.bookingAPI.saveBooking({ bookingDetail: this.form.value }));
    }
  }

  get formControls(): any {
    return this.form.controls;
  }
}
