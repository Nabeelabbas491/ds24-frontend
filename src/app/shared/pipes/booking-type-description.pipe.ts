import { Pipe, PipeTransform } from '@angular/core';
import { MeetingType, MeetingTypeDescriptionType } from './../../types/misc.type';
import { MeetingTypeId } from './../../types/booking.type';

@Pipe({
  name: 'bookingTypeDescription',
})
export class BookingTypeDescriptionPipe implements PipeTransform {
  transform(meetingType: MeetingType, descriptionType: MeetingTypeDescriptionType = 'default'): string {
    let description = '';
    if (descriptionType === 'default') {
      switch (meetingType.id) {
        case MeetingTypeId.Inbound:
          description = 'VENDOR.BOOKING.INBOUND_DESCRIPTION';
          break;
        case MeetingTypeId.Outbound:
          description = 'VENDOR.BOOKING.OUTBOUND_DESCRIPTION';
          break;
        case MeetingTypeId.Zoom:
          description = 'VENDOR.BOOKING.VIDEO_DESCRIPTION';
          break;
      }
    } else {
      switch (meetingType.id) {
        case MeetingTypeId.Inbound:
          description = 'VENDOR_SETTINGS.BOOKING_TYPE.INBOUND_PHONE_CALL.DESCRIPTION';
          break;
        case MeetingTypeId.Outbound:
          description = 'VENDOR_SETTINGS.BOOKING_TYPE.OUTBOUND_PHONE_CALL.DESCRIPTION';
          break;
        case MeetingTypeId.Zoom:
          description = 'VENDOR_SETTINGS.BOOKING_TYPE.VIDEO_CONFERENCE.DESCRIPTION';
          break;
      }
    }
    return description;
  }
}
