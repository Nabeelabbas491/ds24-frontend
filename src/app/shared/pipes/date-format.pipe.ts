import { Pipe, PipeTransform } from '@angular/core';
import { DateFormat } from './../../types/misc.type';
import { TranslateService } from '@ngx-translate/core';
import { format, parseISO } from 'date-fns';

@Pipe({
  name: 'dateFormat',
})
export class DateFormatPipe implements PipeTransform {
  constructor(private translateService: TranslateService) {}
  transform(dateString: string | undefined | null, formatStr: DateFormat = 'full'): string {
    if (dateString == null) {
      return '';
    }
    if (formatStr === 'full') {
      return format(parseISO(dateString), this.translateService.instant('CALENDAR.DAY_FORMAT_STRING.FULL'));
    }
    if (formatStr === 'full_without_year') {
      return format(
        parseISO(dateString),
        this.translateService.instant('CALENDAR.DAY_FORMAT_STRING.FULL_WITHOUT_YEAR'),
      );
    }

    if (formatStr === 'day_month_year') {
      return format(parseISO(dateString), 'dd MMMM, yyyy');
    }
    if (formatStr === 'day_half-month_year') {
      return format(parseISO(dateString), 'dd MMM, yyyy');
    }

    return '';
  }
}
