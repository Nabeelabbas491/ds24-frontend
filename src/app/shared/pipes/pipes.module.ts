import { NgModule } from '@angular/core';

import { AddCommasPipe } from './add-commas.pipe';
import { EllipsisPipe } from './ellipsis.pipe';
import { DateFormatPipe } from './date-format.pipe';
import { TimeSlotPipe } from './time-slot.pipe';
import { BookingTypeDescriptionPipe } from './booking-type-description.pipe';
import { TimezoneFormatPipe } from './timezone-format.pipe';
import { ShowPaginationPipe } from './show-pagination.pipe';

export const PIPES = [
  AddCommasPipe,
  EllipsisPipe,
  DateFormatPipe,
  TimeSlotPipe,
  BookingTypeDescriptionPipe,
  TimezoneFormatPipe,
  ShowPaginationPipe,
];

@NgModule({
  declarations: PIPES,
  exports: PIPES,
})
export class PipesModule {}
