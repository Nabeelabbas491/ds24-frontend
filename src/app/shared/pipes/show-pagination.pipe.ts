import { Pipe, PipeTransform } from '@angular/core';
import { PageBase } from 'src/app/types/misc.type';
import { showPagination } from '../common/util';

@Pipe({
  name: 'showPagination',
})
export class ShowPaginationPipe implements PipeTransform {
  transform(pageBase: PageBase): boolean {
    return showPagination(pageBase);
  }
}
