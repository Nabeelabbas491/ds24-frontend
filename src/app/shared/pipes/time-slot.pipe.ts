import { Pipe, PipeTransform } from '@angular/core';
import { Slot } from './../../types/calendar-day.type';

@Pipe({
  name: 'timeSlot',
})
export class TimeSlotPipe implements PipeTransform {
  transform(slot: Slot | null): string {
    if (slot !== null) {
      return `${slot.startTime} - ${slot.endTime}`;
    }

    return '';
  }
}
