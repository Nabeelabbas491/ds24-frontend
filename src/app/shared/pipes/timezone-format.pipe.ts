import { Pipe, PipeTransform } from '@angular/core';
import { TimeZone } from '../../types/timezone.type';

@Pipe({
  name: 'timezoneFormat',
})
export class TimezoneFormatPipe implements PipeTransform {
  transform(timeZone: TimeZone): string {
    if (timeZone) {
      return timeZone.name + ', ' + timeZone.abbreviatedTimeZone + ' ' + timeZone.offset;
    }
    return '';
  }
}
