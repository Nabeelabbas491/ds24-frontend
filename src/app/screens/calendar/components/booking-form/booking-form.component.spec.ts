import { ComponentFixture, TestBed } from '@angular/core/testing';
import { BookingFormComponent } from './booking-form.component';
import { TranslateModule } from '@ngx-translate/core';
import { DsFormsModule, IconModule } from '@ds24/elements';
import { FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { PipesModule } from '../../../../shared/pipes/pipes.module';
import { State, actions, selectors } from './../../../../store';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { MemoizedSelector } from '@ngrx/store';

describe('BookingFormComponent', () => {
  let fixture: ComponentFixture<BookingFormComponent>;
  let instance: BookingFormComponent;
  let mockStore: MockStore<State>;
  let selectBookingError: MemoizedSelector<State, string | null>;
  let selectBookingPending: MemoizedSelector<State, boolean>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [TranslateModule.forRoot(), IconModule, PipesModule, DsFormsModule, FormsModule, ReactiveFormsModule],
      declarations: [BookingFormComponent],
      schemas: [NO_ERRORS_SCHEMA],
      providers: [
        provideMockStore({
          selectors: [{ selector: selectors.booking.selectMeetingType, value: { id: 2, name: 'Outbound Phone Call' } }],
        }),
      ],
    });
    fixture = TestBed.createComponent(BookingFormComponent);
    instance = fixture.componentInstance;
    mockStore = TestBed.inject(MockStore);
    selectBookingError = mockStore.overrideSelector(selectors.bookingAPI.selectBookingSaveError, null);
    selectBookingPending = mockStore.overrideSelector(selectors.bookingAPI.selectBookingSavePending, false);

    jest.spyOn(mockStore, 'dispatch');
  });

  it('should compile', () => {
    fixture.componentInstance.ngOnInit();
    expect(fixture).toMatchSnapshot();
  });

  it('should disable the form if pending', () => {
    selectBookingPending.setResult(true);
    mockStore.refreshState();
    fixture.componentInstance.ngOnInit();

    expect(fixture).toMatchSnapshot();
  });

  it('should display an error message if provided', () => {
    selectBookingError.setResult('Invalid ');
    mockStore.refreshState();
    fixture.componentInstance.ngOnInit();

    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch an action if the form is valid when submitted', () => {
    const bookingDetail = {
      name: 'Test User',
      email: 'test@test.com',
      phoneNo: '+123456',
      note: 'test test',
    };

    instance.form.setValue(bookingDetail);

    instance.confirmBooking();

    expect(mockStore.dispatch).toHaveBeenCalledWith(actions.bookingAPI.saveBooking({ bookingDetail }));
  });

  it('should not dispatch an action if the form is invalid when submitted', () => {
    const bookingDetail = {
      name: '',
      email: 'test@test.com',
      phoneNo: '123-456',
      note: 'test test',
    };

    instance.form.setValue(bookingDetail);

    instance.confirmBooking();

    expect(mockStore.dispatch).toBeCalledTimes(0);
  });
  it('should dispatch prev step action', () => {
    instance.prevStep();

    const action = actions.booking.selectStep({ step: 'booking_type' });

    expect(mockStore.dispatch).toHaveBeenLastCalledWith(action);
  });
  it('should listen to valueChanges and update leftCountName$', (done: any) => {
    fixture.componentInstance.ngOnInit();
    const name = 'Test Name';
    instance.form.get('name')?.setValue(name);
    instance.leftCountName$.subscribe(value => {
      expect(value).toBe(Math.max(0, 80 - name.length));
      done();
    });
  });

  it('should listen to valueChanges and update leftCountComments$', (done: any) => {
    fixture.componentInstance.ngOnInit();
    const testComments = 'Test Comments';
    instance.form.get('note')?.setValue(testComments);
    instance.leftCountComments$.subscribe(value => {
      expect(value).toBe(Math.max(0, 300 - testComments.length));
      done();
    });
  });

  it('if meetingType is outbound, validations are applied on phoneNumber', (done: any) => {
    fixture.componentInstance.ngOnInit();
    instance.meetingType$.subscribe(() => {
      expect(instance.form.get('phoneNo')?.hasValidator(Validators.required)).toBe(true);
      done();
    });
  });
  it('if meetingType is outbound, make sure pattern validation is working on phone number', (done: any) => {
    fixture.componentInstance.ngOnInit();
    instance.meetingType$.subscribe(() => {
      const bookingDetail = {
        name: 'test user',
        email: 'test@test.com',
        phoneNo: 'abc',
        note: 'test test',
      };

      instance.form.setValue(bookingDetail);

      instance.confirmBooking();

      expect(mockStore.dispatch).toBeCalledTimes(0);
      done();
    });
  });
});
