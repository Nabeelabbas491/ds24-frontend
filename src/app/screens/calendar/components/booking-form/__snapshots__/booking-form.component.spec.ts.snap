// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingFormComponent should compile 1`] = `
<ds-booking-form
  _document={[Function Document]}
  error$={[Function _Store]}
  fb={[Function _FormBuilder]}
  form={[Function FormGroup]}
  isMobileView$={[Function _Store]}
  leftCountComments$={[Function BehaviorSubject]}
  leftCountName$={[Function BehaviorSubject]}
  meetingType$={[Function _Store]}
  meetingTypeIds={[Function Object]}
  ngAfterContentChecked$={[Function Subject]}
  ngAfterContentInit$={[Function Subject]}
  ngAfterViewChecked$={[Function Subject]}
  ngAfterViewInit$={[Function Subject]}
  ngDoCheck$={[Function Subject]}
  ngOnChanges$={[Function Subject]}
  ngOnDestroy$={[Function Subject]}
  ngOnInit$={[Function Subject]}
  saveBookingPending$={[Function _Store]}
  spinnerColor={[Function String]}
  store={[Function _MockStore]}
>
  <div>
    <h4
      class="font-medium"
    >
    </h4>
    <form
      novalidate=""
    >
      <ds-input-wrapper
        class="mt-2 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-input-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <div
            class="flex items-center"
          >
          </div>
        </div>
      </ds-input-wrapper>
      <div
        class="w-full text-left"
      >
      </div>
      <ds-input-wrapper
        class="mt-2 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-input-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <div
            class="flex items-center"
          >
          </div>
        </div>
      </ds-input-wrapper>
      <div
        class="w-full text-left"
      >
      </div>
      <div
        class="w-full text-left"
      >
      </div>
      <ds-textarea
        class="mt-4 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-textarea-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <textarea
            class="h-[150px]"
            dsinput=""
            formcontrolname="note"
            type="_type"
          />
        </div>
      </ds-textarea>
      <div
        class="w-full text-left"
      >
      </div>
      <div
        class="mt-5"
      >
        <button
          class="w-full h-10 py-2 px-4 bg-primary-p300-main text-neutral-n0 shadow-shadow-z200 text-center"
          type="submit"
        >
        </button>
      </div>
      <div
        class="w-full text-center mt-5"
      >
      </div>
    </form>
    <div
      class="flex flex-row justify-center"
    >
    </div>
  </div>
</ds-booking-form>
`;

exports[`BookingFormComponent should disable the form if pending 1`] = `
<ds-booking-form
  _document={[Function Document]}
  error$={[Function _Store]}
  fb={[Function _FormBuilder]}
  form={[Function FormGroup]}
  isMobileView$={[Function _Store]}
  leftCountComments$={[Function BehaviorSubject]}
  leftCountName$={[Function BehaviorSubject]}
  meetingType$={[Function _Store]}
  meetingTypeIds={[Function Object]}
  ngAfterContentChecked$={[Function Subject]}
  ngAfterContentInit$={[Function Subject]}
  ngAfterViewChecked$={[Function Subject]}
  ngAfterViewInit$={[Function Subject]}
  ngDoCheck$={[Function Subject]}
  ngOnChanges$={[Function Subject]}
  ngOnDestroy$={[Function Subject]}
  ngOnInit$={[Function Subject]}
  saveBookingPending$={[Function _Store]}
  spinnerColor={[Function String]}
  store={[Function _MockStore]}
>
  <div>
    <h4
      class="font-medium"
    >
    </h4>
    <form
      novalidate=""
    >
      <ds-input-wrapper
        class="mt-2 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-input-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <div
            class="flex items-center"
          >
          </div>
        </div>
      </ds-input-wrapper>
      <div
        class="w-full text-left"
      >
      </div>
      <ds-input-wrapper
        class="mt-2 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-input-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <div
            class="flex items-center"
          >
          </div>
        </div>
      </ds-input-wrapper>
      <div
        class="w-full text-left"
      >
      </div>
      <div
        class="w-full text-left"
      >
      </div>
      <ds-textarea
        class="mt-4 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-textarea-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <textarea
            class="h-[150px]"
            dsinput=""
            formcontrolname="note"
            type="_type"
          />
        </div>
      </ds-textarea>
      <div
        class="w-full text-left"
      >
      </div>
      <div
        class="mt-5"
      >
        <button
          class="w-full h-10 py-2 px-4 bg-primary-p300-main text-neutral-n0 shadow-shadow-z200 text-center"
          type="submit"
        >
        </button>
      </div>
      <div
        class="w-full text-center mt-5"
      >
      </div>
    </form>
    <div
      class="flex flex-row justify-center"
    >
    </div>
  </div>
</ds-booking-form>
`;

exports[`BookingFormComponent should display an error message if provided 1`] = `
<ds-booking-form
  _document={[Function Document]}
  error$={[Function _Store]}
  fb={[Function _FormBuilder]}
  form={[Function FormGroup]}
  isMobileView$={[Function _Store]}
  leftCountComments$={[Function BehaviorSubject]}
  leftCountName$={[Function BehaviorSubject]}
  meetingType$={[Function _Store]}
  meetingTypeIds={[Function Object]}
  ngAfterContentChecked$={[Function Subject]}
  ngAfterContentInit$={[Function Subject]}
  ngAfterViewChecked$={[Function Subject]}
  ngAfterViewInit$={[Function Subject]}
  ngDoCheck$={[Function Subject]}
  ngOnChanges$={[Function Subject]}
  ngOnDestroy$={[Function Subject]}
  ngOnInit$={[Function Subject]}
  saveBookingPending$={[Function _Store]}
  spinnerColor={[Function String]}
  store={[Function _MockStore]}
>
  <div>
    <h4
      class="font-medium"
    >
    </h4>
    <form
      novalidate=""
    >
      <ds-input-wrapper
        class="mt-2 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-input-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <div
            class="flex items-center"
          >
          </div>
        </div>
      </ds-input-wrapper>
      <div
        class="w-full text-left"
      >
      </div>
      <ds-input-wrapper
        class="mt-2 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-input-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <div
            class="flex items-center"
          >
          </div>
        </div>
      </ds-input-wrapper>
      <div
        class="w-full text-left"
      >
      </div>
      <div
        class="w-full text-left"
      >
      </div>
      <ds-textarea
        class="mt-4 text-left"
      >
        <div
          class="flex flex-col"
        >
          <ds-form-label-wrapper
            class="truncate"
            data-test-textarea-label=""
          >
            <label>
              <ds-form-label>
              </ds-form-label>
            </label>
          </ds-form-label-wrapper>
          <textarea
            class="h-[150px]"
            dsinput=""
            formcontrolname="note"
            type="_type"
          />
        </div>
      </ds-textarea>
      <div
        class="w-full text-left"
      >
      </div>
      <div
        class="mt-5"
      >
        <button
          class="w-full h-10 py-2 px-4 bg-primary-p300-main text-neutral-n0 shadow-shadow-z200 text-center"
          type="submit"
        >
        </button>
      </div>
      <div
        class="w-full text-center mt-5"
      >
      </div>
    </form>
    <div
      class="flex flex-row justify-center"
    >
    </div>
  </div>
</ds-booking-form>
`;
